ppx/reason_react_ppx.ml

```diff
@@ -45,16 +45,163 @@ let merlinFocus =
+            [];
+        ];
+    attr_loc = loc;
+  }
+
+let createWarningSuppressionAttribute ~loc =
+  {
+    attr_name = { txt = "warning"; loc };
+    attr_payload =
+      PStr
+        [
+          Builder.pstr_eval ~loc
+            (Builder.pexp_constant ~loc (Pconst_string ("-32", loc, None)))
+            [];
+        ];
+    attr_loc = loc;
+  }
+
+let rec buildArrowType ~loc props =
+  match props with
+  | [] ->
+      Builder.ptyp_arrow ~loc Nolabel
+        (Builder.ptyp_constr ~loc { txt = Lident "unit"; loc } [])
+        (Builder.ptyp_var ~loc "a")
+  | (label, _) :: rest ->
+      let propName = getLabelOrEmpty label in
+      let propType =
+        if propName = "children" then
+          Builder.ptyp_constr ~loc
+            { txt = Ldot (Lident "React", "element"); loc }
+            []
+        else if propName = "style" then
+          Builder.ptyp_constr ~loc
+            { txt = Ldot (Ldot (Lident "ReactDOM", "Style"), "t"); loc }
+            []
+        else if propName = "onClick" then
+          Builder.ptyp_arrow ~loc Nolabel
+            (Builder.ptyp_var ~loc "event")
+            (Builder.ptyp_constr ~loc { txt = Lident "unit"; loc } [])
+        else if propName = "onChange" then
+          Builder.ptyp_arrow ~loc Nolabel
+            (Builder.ptyp_var ~loc "event")
+            (Builder.ptyp_constr ~loc { txt = Lident "unit"; loc } [])
+        else Builder.ptyp_constr ~loc { txt = Lident "string"; loc } []
+      in
+      let finalLabel, propType' =
+        if isDataProp label then
+          let jsName = transformToKebabCase propName in
+          let melAsAttr = createMelAsAttribute ~loc jsName in
+          (Labelled propName, { propType with ptyp_attributes = [ melAsAttr ] })
+        else (Labelled propName, propType)
+      in
+      Builder.ptyp_arrow ~loc finalLabel propType' (buildArrowType ~loc rest)
+
+let createExternalDeclaration ~name ~props ~loc =
+  {
+    pstr_desc =
+      Pstr_primitive
+        {
+          pval_name = { txt = name; loc };
+          pval_type = buildArrowType ~loc props;
+          pval_prim = [ "" ];
+          (* Empty string for [@mel.obj] *)
+          pval_attributes =
+            [
+              createMelObjAttribute ~loc; 
+              createWarningSuppressionAttribute ~loc;
+              List.hd merlinHideAttrs;
+            ];
+          pval_loc = loc;
+        };
+    pstr_loc = loc;
+  }
 
 module Binding = struct
   (* Binding is the interface that the ppx relies on to interact with the react bindings.
      Here we define the same APIs as the bindings but it generates Parsetree nodes *)
   module ReactDOM = struct
-    let domProps ~applyLoc ~loc props =
-      Builder.pexp_apply ~loc:applyLoc
-        (Builder.pexp_ident ~loc:applyLoc ~attrs:merlinHideAttrs
-           { loc; txt = Ldot (Lident "ReactDOM", "domProps") })
-        props
+    let domProps ~applyLoc ~loc ?(elementName = "element") props =
```

Comment:
This module is just the API of the reason-react to be accesssed into the ppx. Don't add logic to it.